### ISSUE TITLE
Otel collector publish traces with OTLPResourceNoServiceName

### ISSUE THREAD

**Issue by Xosmond** [jgaskins/opentelemetry#2]
When using Otel-collector, it publishes results with service name as `OTLPResourceNoServiceName` even when I specified service_name.

```
OpenTelemetry.configure do |c|
  c.service_name = "crystallove"
  c.exporter = OpenTelemetry::BatchExporter.new(
    OpenTelemetry::HTTPExporter.new(
      endpoint: URI.parse("http://0.0.0.0:4318"),
      headers: HTTP::Headers{
        "content-type" => "application/x-protobuf",
      },
    )
  )
end
```

I will try to debug someway to find what could be the reason but maybe you know the issue here.

**Comment by jgaskins** [comment]
Hmm, this is one I don't think I've seen. The convention is to [use the `service.name` attribute](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/README.md#service), which is set [here](https://github.com/jgaskins/opentelemetry/blob/ca32f4a72f0e1fc67d6f2e3a5f4ee90176b08e0e/src/opentelemetry.cr#L41).

It's possible that it's a "resource" thing, which may be different from "traces"? Honeycomb and Datadog inferred it okay from `service.name` attribute of the top-level span of a trace, but maybe other implementations require a legit implementation of OTel Resources. I don't know where that goes in the payload yet, though.

**Comment by Xosmond** [comment]
I'm gonna close this. Looks like it depends on the consumer/viewer. 

I this case I was trying to use "openzipkin/zipkin" /  "jaegertracing/all-in-one".

They take the service_name inside tags

**Comment by jgaskins** [comment]
Ah, okay. If it’s something this shard can accommodate, feel free to
reopen.

On November 2, 2021, GitHub ***@***.***> wrote:
> Closed #2 <https://github.com/jgaskins/opentelemetry/issues/2>.
>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/jgaskins/opentelemetry/issues/2#event-5559146298>,
or unsubscribe <https://github.com/notifications/unsubscribe-
auth/AAA2NLO3T5TGAUW4FDTHTGDUKCLNHANCNFSM5HFKFM5Q>.
Triage notifications on the go with GitHub Mobile for iOS
<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-
email&mt=8&pt=524675> or Android
<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-
email%26utm_medium%3Demail%26utm_source%3Dgithub>.
